Kronos-Shifts-Connector/Microsoft.Teams.Shifts.Integration/Microsoft.Teams.Shifts.Integration.API/Controllers/TeamsController.cs

```diff
@@ -115,6 +115,12 @@ public async Task<HttpResponseMessage> CheckSetupAsync()
         [Route("/v1/teams/{aadGroupId}/update")]
         public async Task<ActionResult> UpdateTeam([FromRoute] string aadGroupId)
         {
+            var request = this.Request;
+            var requestHeaders = request.Headers;
+            Microsoft.Extensions.Primitives.StringValues passThroughValue = string.Empty;
+
+            requestHeaders.TryGetValue("X-MS-WFMPassthrough", out passThroughValue);
```

Comment:
Short-circuit processing here, if you detect that it's a call made by your integration.

Reply:
@aosolis , We wanted to implement passthrough header for OpenShiftRequest and SwapShiftRequest only. As in other cases (Shifts and Open Shifts) we have to simply send the response as it can be a genuine request from Shifts's other entity.